### PR TITLE
[feat] Link user signup to the database 

### PR DESCRIPTION
**Pull request by homosapientdream**
## Description

- added the CRUD logic in user controller to add users create, get, and update the user database
- connect the signup and login page to the user database via modifying login/signup.tsx and server.js 

## Reproduction steps

- to test: signup as a new user, link with the database in the environment and select all from users table. 
- if the signed up user info is in the database, then the test is done. 

## Checklist

- [x] I've followed the [Contributing guidelines](https://github.com/oslabs-beta/.github/blob/main/docs/CONTRIBUTING.md)
- [x] I've titled my PR according to the [Conventional Commits spec](https://conventionalcommits.org)
- [ ] I've added tests that fail without this PR but pass with it
- [ ] I've linted, tested, and commented my code
- [ ] I've updated documentation (if appropriate)

### CODE REVIEW

File: server/controllers/userController.js [hunk 1]
```diff
@@ -1,14 +1,113 @@
...
+      console.error('error logging in user', err);
+      res
+        .status(500)
+        .json({ error: 'login error in server (userController page)' });
+    } finally {
+      client.release();
+    }
+  },
+
+  getAllUsers: async (req, res) => {
```
Comment (aidanprior) [inline server/controllers/userController.js:81]: For security reasons, we probably shouldn't expose this middleware function to any route, as a user could query all other users and get personal account information

File: server/controllers/userController.js [hunk 2]
```diff
@@ -1,14 +1,113 @@
...
+      const result = await client.query(queryText);
+      res.status(200).json(result.rows);
+    } catch (err) {
+      console.error('Error fetching users:', err);
+    } finally {
+      client.release();
+    }
+  },
+
+  getUserByField: async (req, res) => {
```
Comment (aidanprior) [inline server/controllers/userController.js:94]: Similar to above: For security reasons, we don't want personal information to be returned here

File: server/server.js [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
+//signup router
+app.post('/api/signup', userController.createUser);
+
+//login router
+app.post('api/login', userController.loginUser);
+
+app.get('/users/');
```
Comment (aidanprior) [inline server/server.js:32]: What is the purpose of this route currently?

File: server/server.js [hunk 4]
```diff
@@ -1,13 +1,36 @@
...
+  })
+  .catch((err) => {
+    console.error('Connection Error in Connecting with database', err.stack); //a property of the Error object that contains a string representing the error message and the stack trace.
+  });
+
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
+//signup router
+app.post('/api/signup', userController.createUser);
```
Comment (aidanprior) [inline server/server.js:27]: We should have a final endpoint handler for each route, and a global error handler

**Comment by aidanprior** [comment]
Also, Nancy, please include the issue number in the pull request description. It should link to our project's card and automate status setting that way.

**Review by SamLarivs — COMMENTED** [review]
Solid, Nancy. Aidans got some really good feedback. Thanks for gettin that going.

**Comment by homosapientdream** [comment]
thanks for your comprehensive feedbacks! Aidan!

**Review by mkw2107 — DISMISSED** [review]

**Review by aidanprior — APPROVED** [review]

**Pull request merged**